src/goto-symex/symex_goto.cpp

```diff
@@ -429,6 +429,14 @@ void goto_symext::phi_function(
       rhs=goto_state_rhs;
     else if(goto_state.guard.is_false())
       rhs=dest_state_rhs;
+    else if(goto_state.level2_current_count(l1_identifier) == 0)
```

Comment:
Add comments briefly explaining why these checks are here

Reply:
Added a comment block at the top of the if.